main.tf

```diff
@@ -211,11 +211,85 @@ resource "aws_autoscaling_group" "container_instance" {
   }
 }
 
+resource "aws_launch_configuration" "container_instance_scheduled_tasks" {
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  root_block_device {
+    volume_type = "${var.root_block_device_type}"
+    volume_size = "${var.root_block_device_size}"
+  }
+
+  name_prefix          = "lc${title(var.environment)}ContainerInstanceScheduledTasks-"
+  iam_instance_profile = "${aws_iam_instance_profile.container_instance.name}"
+
+  # Using join() is a workaround for depending on conditional resources.
+  # https://github.com/hashicorp/terraform/issues/2831#issuecomment-298751019
+  image_id = "${var.lookup_latest_ami ? join("", data.aws_ami.ecs_ami.*.image_id) : join("", data.aws_ami.user_ami.*.image_id)}"
+
+  instance_type   = "${var.instance_type}"
+  key_name        = "${var.key_name}"
+  security_groups = ["${aws_security_group.container_instance.id}"]
+  user_data       = "${data.template_cloudinit_config.container_instance_cloud_config.rendered}"
+}
+
+resource "aws_autoscaling_group" "container_instance_scheduled_tasks" {
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  name                      = "asg${title(var.environment)}ContainerInstanceScheduledTasks"
+  launch_configuration      = "${aws_launch_configuration.container_instance_scheduled_tasks.name}"
+  health_check_grace_period = "${var.health_check_grace_period}"
+  health_check_type         = "EC2"
+  desired_capacity          = "${var.desired_capacity_scheduled_tasks}"
+  termination_policies      = ["OldestLaunchConfiguration", "Default"]
+  min_size                  = "${var.min_size_scheduled_tasks}"
+  max_size                  = "${var.max_size_scheduled_tasks}"
+  enabled_metrics           = ["${var.enabled_metrics}"]
+  vpc_zone_identifier       = ["${var.private_subnet_ids}"]
+
+  tag {
+    key                 = "Name"
+    value               = "ContainerInstanceScheduledTasks"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "Project"
+    value               = "${var.project}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "Environment"
+    value               = "${var.environment}"
+    propagate_at_launch = true
+  }
+}
+
 #
 # ECS resources
 #
 resource "aws_ecs_cluster" "container_instance" {
   name = "ecs${title(var.environment)}Cluster"
+  capacity_providers = [aws_ecs_capacity_provider.scheduled_tasks.name]
+  default_capacty_provider_strategy {
+    capacity_provider = aws_ecs_capacity_provider.scheduled_tasks.name
+  }
+}
+
+resource "aws_ecs_capacity_provider" "scheduled_tasks" {
```

Comment:
I think it should be named more generically so that it makes sense for it to be used for things beyond scheduled tasks.

Reply:
I don't have any plans to use this capacity provider for anything else besides scheduled tasks. There could be multiple capacity providers for a cluster, if that helps make this make more sense.

Reply:
What benefit is there to multiple capacity providers? Downside is it reduces resource utilization. Example, you have capacity providers A, B, C. Capacity provider A was used to bring up an extra EC2 instance, but that EC2 instance is not fully utilized. Now a service using capacity provider B is trying to launch, but B doesn't have resources (A does, but it's not using A), so we bring up an unnecessary instance on B.

If we had something special about these instances, for example, we used a different instance type or size that was very different I could see some benefits. Like if you wanted to use GPU instances or had some odd tasks that required a huge amount of resources, so you only want to bring up say m5.16xlarge just for those specific tasks.

This PR sets the "scheduled tasks" capacity provider was set as the default capacity provider for the cluster. The default should have a generic name.

So in summary, we should have a generic default capacity provider and it should be named as such. If we have a need for special capacity providers, then we could create them as needed. I don't see a need and this is the default, so I think it should have a generic name (or not be defined as the default).

Reply:
Having this be the "default" provider is the easiest way for us to get scheduled tasks on a different instance type. This is probably incompatible with a more general use of capacity providers, so I would expect more extending changes if we are to use capacity providers for more than scheduled tasks.